Library/Homebrew/formula_subset_printer.rb

```diff
@@ -0,0 +1,72 @@
+require "formula"
+require "tab"
+require "diagnostic"
+
+module Homebrew
+  class FormulaSubsetPrinter
+    attr_reader :universe, :subset
+
+    def initialize
+      @universe = universe
+      @subset = subset
+    end
+
+    def subset
+      raise "method missing"
+    end
+
+    def universe
+      if ARGV.named.empty?
+        Formula.installed
+      else
+        ARGV.resolved_formulae
+      end
+    end
+
+    def self.print
+      return unless HOMEBREW_CELLAR.exist?
+      printer = new
+      if ARGV.json == "v1"
+        printer.print_json
+      elsif ($stdout.tty? || ARGV.verbose?) && !ARGV.flag?("--quiet")
+        printer.print_verbose
+      else
+        printer.print_quiet
+      end
+      printer.exit_status
+    end
+
+    def exit_status
+      Homebrew.failed = @subset.any?
+    end
+
+    def quiet
+      @subset.class == Hash ? @subset.values.flatten.uniq : @subset
```

Comment:
Alternatively, it could be either
- abstract and always implemented in subclass

or

- defined here as `@subset` by default, and then overridden in `missing` as `@subset.values.flatten.uniq`